modules/auxiliary/scanner/ssh/ssh_version.rb

```diff
@@ -4,96 +4,219 @@
 ##
 
 require 'recog'
+require 'net/ssh/transport/session'
 
 class MetasploitModule < Msf::Auxiliary
-  include Msf::Exploit::Remote::Tcp
   include Msf::Auxiliary::Scanner
   include Msf::Auxiliary::Report
 
-  # the default timeout (in seconds) to wait, in total, for both a successful
-  # connection to a given endpoint and for the initial protocol response
-  # from the supposed SSH endpoint to be returned
-  DEFAULT_TIMEOUT = 30
-
   def initialize
     super(
-      'Name'        => 'SSH Version Scanner',
-      'Description' => 'Detect SSH Version.',
-      'References'  =>
-        [
-          [ 'URL', 'https://en.wikipedia.org/wiki/SecureShell' ]
-        ],
-      'Author'      => [ 'Daniel van Eeden <metasploit[at]myname.nl>' ],
-      'License'     => MSF_LICENSE
+      'Name' => 'SSH Version Scanner',
+      'Description' => 'Detect SSH Version, and the algorithms available from the server',
+      'References' => [
+        ['URL', 'https://en.wikipedia.org/wiki/SecureShell'], # general info
+        ['URL', 'https://datatracker.ietf.org/doc/html/rfc8732#name-deprecated-algorithms'], # deprecation of kex gss-sha1 stuff
+        ['URL', 'https://datatracker.ietf.org/doc/html/draft-ietf-curdle-ssh-kex-sha2-20#page-16'], # diffie-hellman-group-exchange-sha1, diffie-hellman-group1-sha1, rsa1024-sha1
+        ['URL', 'https://datatracker.ietf.org/doc/html/rfc8758#name-iana-considerations'], # arc4 deprecation
+        ['URL', 'https://github.com/net-ssh/net-ssh?tab=readme-ov-file#supported-algorithms'], # a bunch of diff removed things from the ruby lib
+        ['CVE', '2008-5161'] # CBC modes
+      ],
+      'Author' => [
+        'Daniel van Eeden <metasploit[at]myname.nl>', # original author
+        'h00die' # algorithms enhancements
+      ],
+      'License' => MSF_LICENSE
     )
 
     register_options(
       [
         Opt::RPORT(22),
-        OptInt.new('TIMEOUT', [true, 'Timeout for the SSH probe', DEFAULT_TIMEOUT])
+        OptInt.new('TIMEOUT', [true, 'Timeout for the SSH probe', 30])
       ],
       self.class
     )
   end
 
   def timeout
-    datastore['TIMEOUT'] <= 0 ? DEFAULT_TIMEOUT : datastore['TIMEOUT']
+    datastore['TIMEOUT']
+  end
+
+  def rport
+    datastore['RPORT']
   end
 
   def run_host(target_host)
     ::Timeout.timeout(timeout) do
-      connect
+      transport = Net::SSH::Transport::Session.new(target_host, { port: rport })
+
+      server_data = transport.algorithms.instance_variable_get(:@server_data)
+      host_keys = transport.algorithms.session.instance_variable_get(:@host_keys).instance_variable_get(:@host_keys)
+      print_status("#{target_host} - Key Fingerprint: #{host_keys[0].fingerprint}") if host_keys.length.positive?
```

Comment:
Alright so there's a couple of things here.

1. We should show all the fingerprints in the array.
2. It'd be _really_ nice if the fingerprint were the same as the output from OpenSSH / `ssh -v` / `ssh_keyscan`. If it's the same, it'd be easier for users to cross reference.

```suggestion
      host_keys = transport.algorithms.session.instance_variable_get(:@host_keys).instance_variable_get(:@host_keys)
      host_keys.each do |host_key|
        print_status("#{target_host} - Key Fingerprint: #{host_key.ssh_type} #{Base64.strict_encode64(host_key.to_blob)}")
      end
```